### PR TITLE
Added new OpMon project to monitor network performance, build by build

### PR DESCRIPTION
**Pull request by acreskeyMoz**
Starting with a handful of metrics, defined in the Networking outcome

### CODE REVIEW

**Comment by acreskeyMoz** [comment]
My first OpMon, please let me know if you see anything amiss.

File: opmon/network-performance-desktop-build-by-build.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+platform = "firefox_desktop"
+xaxis = "build_id"
+start_date = "2023-01-01"
+metrics = [
+    "dns_lookup_time",
+    "http_page_tls_handshake",
+    "http_sub_tls_handshake",
+    "http_page_open_to_first_sent",
+    "time_to_response_start_ms"
+]
```
Comment (acreskeyMoz) [inline opmon/network-performance-desktop-build-by-build.toml:15]: These are defined here: [jetstream/outcomes/firefox_desktop/networking.toml](https://github.com/mozilla/metric-hub/blob/main/jetstream/outcomes/firefox_desktop/networking.toml)

Does that work?

Comment (scholtzan) [reply opmon/network-performance-desktop-build-by-build.toml:15]: Unfortunately, no. Outcomes basically define a set of metrics that should always be computed together in Jetstream. Outcomes are currently not supported in OpMon (open issue https://github.com/mozilla/opmon/issues/153)

Generally, to reference metrics by their slugs, they'll need to be added as a definition in here: https://github.com/mozilla/metric-hub/tree/main/definitions

We are currently working on https://github.com/mozilla/jetstream/issues/1629 which will make it possible to reference these metric definitions in outcomes (which currently does not work). Once that is implemented you could move the metric definitions out of the outcome definition into a [definitions file](https://github.com/mozilla/metric-hub/blob/main/definitions/firefox_desktop.toml) and then simply reference the metrics in the outcome and in this config file. However, it'll probably take a week or two until we implement that support.

Until then, I'd suggest as a quick workaround to copy-paste the metric definitions into this config file (and then later simply delete them once they live in definitions).

File: opmon/network-performance-desktop-build-by-build.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+skip_default_metrics = true
+
+[project.population]
+
+data_source = "main"
+channel = "beta"
+dimensions = ["os"]
+monitor_entire_population = true
+
```
Comment (scholtzan) [inline opmon/network-performance-desktop-build-by-build.toml:24]: The duplication of definitions is certainly not great, but the quickest way to get results until the linked issue gets implemented. 

Also, OpMon offers different statistics that are less computationally expensive (jetstream is very expensive, and OpMon is supposed to serve as a more lightweight, cheaper sibling to it). Statistics that are available are:
* Mean
* Count
* Sum
* Quantile
* Percentile
* Total ratio

Feel free to change the statistics, and/or add more. For now I selected percentiles for the metrics.

```suggestion
[metrics.dns_lookup_time]
select_expression = '{{agg_histogram_mean("payload.processes.content.histograms.dns_lookup_time")}}'
data_source = 'main'
bigger_is_better = false

[metrics.dns_lookup_time.statistics]
percentile = { percentiles = [25, 50, 75, 95] }


[metrics.http_page_tls_handshake]
select_expression = '{{agg_histogram_mean("payload.processes.content.histograms.http_page_tls_handshake")}}'
data_source = 'main'
bigger_is_better = false

[metrics.http_page_tls_handshake.statistics]
percentile = { percentiles = [25, 50, 75, 95] }


[metrics.http_sub_tls_handshake]
select_expression = '{{agg_histogram_mean("payload.processes.content.histograms.http_sub_tls_handshake")}}'
data_source = 'main'
bigger_is_better = false

[metrics.http_sub_tls_handshake.statistics]
percentile = { percentiles = [25, 50, 75, 95] }


[metrics.http_page_open_to_first_sent]
select_expression = '{{agg_histogram_mean("payload.processes.content.histograms.http_page_open_to_first_sent")}}'
data_source = 'main'
bigger_is_better = false

[metrics.http_page_open_to_first_sent.statistics]
percentile = { percentiles = [25, 50, 75, 95] }


[metrics.time_to_response_start_ms]
select_expression = '{{agg_histogram_mean("payload.processes.content.histograms.time_to_response_start_ms")}}'
data_source = 'main'
bigger_is_better = false

[metrics.time_to_response_start_ms.statistics]
percentile = { percentiles = [25, 50, 75, 95] }
```

Comment (acreskeyMoz) [reply opmon/network-performance-desktop-build-by-build.toml:24]: Thank you @scholtzan , I've added your suggested presentation to the patch.

**Review by scholtzan — APPROVED** [review]

**Pull request merged**